modules/telemetry/json_packets.py

```diff
@@ -185,3 +188,167 @@ def parse_mission_file(mission_file: Path) -> MissionEntry:
             length += 1
 
     return MissionEntry(name=mission_file.stem, length=length, filepath=mission_file, version=1)
+
+
+@dataclass
+class TelemetryDataPacketBlock:
+    """A generic block object to store information for telemetry data
+    All stored values must be updated at once!"""
+
+    mission_time: list[int] = field(default_factory=list[int])
+    stored_values: dict[str, list[int]] = field(default_factory=dict)
+
+    def __init__(self, keys: list[str]):
+        """Initializes a TelemetryDataPacketBlock"""
+        self.mission_time = []
+        self.stored_values = dict.fromkeys(keys, [])
+        logger.debug(self)
+
+    def update(self, data: dict[str, int], buffer_size: int) -> None:
+        """Updates the stored values with the given data
+        :param data: Dictionary of data to update containing mission_time and stored_values squashed
+        :param buffer_size: Size of the telemetry buffer to store"""
+        # Ensure you are not half updating the packet
+        # As this can cause the arrays to become out of sync and meaningless.
+        if dict(self).keys() != data.keys():
```

Comment:
Are we sure this works? Won't `dict(self)` include method names?

Reply:
Before I override __iter__ it only returns  "mission_time" and all other stored variables. For example,

["mission_time", "celsius"]

Debug printing them on every update() returns this:
DEBUG:modules.telemetry.json_packets:UPDATING dict_keys(['mission_time', 'celsius']) using dict_keys(['celsius', 'mission_time'])!

Reply:
Right, I was thinking of `self.__dict__`